src/planner/planner.c

```diff
@@ -553,11 +570,21 @@ timescaledb_planner(Query *parse, int cursor_opts, ParamListInfo bound_params)
 				 * or portal memory contexts could also be suitable, but they
 				 * don't exist for SPI calls.
 				 */
-				MemoryContextStats(CurrentMemoryContext);
 				ts_baserel_info = BaserelInfo_create(CurrentMemoryContext,
 													 /* nelements = */ 1,
 													 /* private_data = */ NULL);
 			}
+
+			/*
+			 * Create a temporary memory context that can optionally be used by
+			 * the planner at this level of timescaledb_planner(). The inner
+			 * recursive invocations of timescaledb_planner() will also create
+			 * their own temporary contexts.
+			 */
+			MemoryContext mcxt = AllocSetContextCreate(CurrentMemoryContext,
```

Comment:
I'm kinda reluctant to add another weird cache based on static variables, and an entire stack this time... On the other hand, the general Postgres planning approach of "leaking memory like there's no tomorrow" ((c) the mailing lists) becomes problematic when you have a lot of partitions.

Reply:
I 'm not a fan of adding nested memory contexts like this.

Reply:
Yeah, I don't like it either, feels fragile. We can just drop this as wontfix. Or maybe you have better ideas?

Reply:
The planning memory usage proportional to the number of chunks times the number of array elements also doesn't feel right, but not sure how to fix it in extension.